akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -355,11 +353,14 @@ object Source extends SourceApply {
    * if there is no demand from downstream. When `bufferSize` is 0 the `overflowStrategy` does
    * not matter.
    *
-   * The stream can be completed successfully by sending [[akka.actor.PoisonPill]] or
-   * [[akka.actor.Status.Success]] to the actor reference.
+   * The stream can be completed successfully by sending the actor referende an [[akka.actor.Status.Success]]
+   * messagein which case already buffered elements will be signalled before signalling completion,
+   * or by sending a [[akka.actor.PoisonPill]] in which case completion will be signalled immediatly.
    *
    * The stream can be completed with failure by sending [[akka.actor.Status.Failure]] to the
-   * actor reference.
+   * actor reference. In case the Actor is still draining its internal buffer (after having received
+   * an [[akka.actor.Status.Success]]) before signalling completion and it receives a [[akka.actor.Status.Failure]],
+   * the failure will be signalled downstream immediatly (instead of the completion signal).
```

Comment:
I guess we should stick to UK spelling or US spelling... We have a mix currently ("ll" is UK)